src/main/java/de/tum/cit/aet/artemis/communication/web/AnswerMessageResource.java

```diff
@@ -92,4 +95,11 @@ public ResponseEntity<Void> deleteAnswerMessage(@PathVariable Long courseId, @Pa
         // deletion of answerMessages should not trigger alert
         return ResponseEntity.ok().build();
     }
+
+    @GetMapping("courses/{courseId}/answer-messages/source-answer-posts")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<AnswerPost>> getSourceAnswerPostsByIds(@PathVariable Long courseId, @RequestParam List<Long> answerPostIds) {
+        List<AnswerPost> answerPosts = answerMessageService.findByIdIn(answerPostIds);
```

Comment:
you are missing a security check that the user is at least student in the course, either change the annoation above or add an additional check here. This should be part of the resource and not be done in a service (except it's clear from the method names)